### PR TITLE
Added test and example for -c/--creator feature

### LINKED ISSUE #131: test, and document `creator` feature
the [-c/--creator](https://jupysql.ploomber.io/en/latest/api/magic-sql.html) feature is not well documented; we only describe what it is, but there are no runnable examples. also, I quickly look at the code and seems that we don't have any unit tests.

[based on this](https://docs.sqlalchemy.org/en/20/core/engines.html#sqlalchemy.create_engine.params.creator), looks like creator is passed to sqlalchemy to initialize a connection. this feature sounds very similar to the one we added recently for connecting to [existing engines](https://jupysql.ploomber.io/en/latest/howto.html#connect-to-existing-engine). I think we should keep both to remain 100% compatible with ipython-sql  but recommend users switch to the new feature.

### PR DESCRIPTION
**Pull request by DaveOkpare**
## Describe your changes

Added test and example for -c/--creator feature

## Issue number

Closes #131 

## Checklist before requesting a review

- [x] Performed a self-review of my code
- [x] Formatted my code with [`pkgmt format`](https://ploomber-contributing.readthedocs.io/en/latest/contributing/pr.html#linting-formatting)
- [x] Added [tests](https://ploomber-contributing.readthedocs.io/en/latest/contributing/pr.html#testing) (when necessary).
- [x] Added [docstring](https://ploomber-contributing.readthedocs.io/en/latest/contributing/pr.html#documenting-changes-and-new-features) documentation and update the [changelog](https://ploomber-contributing.readthedocs.io/en/latest/contributing/pr.html#changelog) (when needed)

----
:books: Documentation preview :books:: https://jupysql--567.org.readthedocs.build/en/567/

### CODE REVIEW

File: doc/api/magic-sql.md [hunk 1]
```diff
@@ -136,6 +136,21 @@ Or pass an alias (**added in 0.5.2**):
 %sql --close db-two
 ```
 
+
+## Specify creator function
```
Comment (edublancas) [inline doc/api/magic-sql.md:140]: according to the [docs](https://docs.sqlalchemy.org/en/20/core/engines.html#sqlalchemy.create_engine.params.creator), creator must be a function that returns a DBAPI connection:

```
In [11]: engine = create_engine("sqlite://", creator=lambda:sqlite3.connect(""))

In [12]: engine.execute('SELECT name FROM sqlite_schema WHERE type=\'table\' ORDER BY name;')
Out[12]: <sqlalchemy.engine.cursor.LegacyCursorResult at 0x103531840>

In [13]: engine.execute('SELECT name FROM sqlite_schema WHERE type=\'table\' ORDER BY name;').fetchall()
Out[13]: []
```

please update the examples and the test.

**Review by edublancas — CHANGES_REQUESTED** [review]

File: doc/api/magic-sql.md (hunk 1 above)
Comment (edublancas) [reply doc/api/magic-sql.md:140]: yeah, I got the same error. probably an issue on duckdb's end, please add an example with sqlite as I showed in the previous comment

**Review by edublancas — CHANGES_REQUESTED** [review]
check your code, it's failing due to a listing problem

**Comment by DaveOkpare** [comment]
> check your code, it's failing due to a linting problem

@edublancas Thanks for the heads up, I made too many commits before reading the linting logs, should I open a new pull request to make things cleaner?

File: doc/api/magic-sql.md [hunk 2]
```diff
@@ -136,6 +136,27 @@ Or pass an alias (**added in 0.5.2**):
...
+
+# Define a function that returns a DBAPI connection
+def creator():
+    return sqlite3.connect("")
+
+%sql --creator creator
+```
+
+```{code-cell} ipython3
+%sql --creator engine
```
Comment (edublancas) [inline doc/api/magic-sql.md:156]: the docs are failing bc the `engine` variable doesn't exist

**Review by edublancas — CHANGES_REQUESTED** [review]
no need to open a new PR. just  ensure all tests pass

**Review by edublancas — APPROVED** [review]

**Comment by edublancas** [comment]
thanks for your contribution! @DaveOkpare

**Pull request merged**